tests/core/test_variable.py

```diff
@@ -0,0 +1,771 @@
+
+
+# 371 - end, no function
+
+class variable__strange_end_attribute(Variable):
+    column = IntCol
+    entity = Individu
+    definition_period = MONTH
+    label = u"Variable with dubious end attribute, no function."
+    end = '1989-00-00'
+
+
+def test_variable__strange_end_attribute():
+    try:
+        tax_benefit_system.add_variable(variable__strange_end_attribute)
+
+    except ValueError, e:
+        if hasattr(e, 'message'):
+            message = getattr(e, 'message')
+            assert message.startswith("Incorrect 'end' attribute format"), message
+    except:
+        raise
+
+    # Check that Error at variable adding prevents it from registration in the taxbenefitsystem.
+    assert not hasattr(tax_benefit_system.column_by_name, "variable__strange_end_attribute")
+
+
+# 371 - end, no function
+
+class variable__dated_attribute(Variable):
+    column = IntCol
+    entity = Individu
+    definition_period = MONTH
+    label = u"Variable with end attribute, no function."
+    # start_date = datetime.date(1980, 1, 1)
+    end = '1989-12-31'
+
+
+tax_benefit_system.add_variable(variable__dated_attribute)
+
+
+def test_variable__dated_attribute():
+    variable = tax_benefit_system.column_by_name['variable__dated_attribute']
+    assert variable is not None
+    assert variable.end == '1989-12-31'
+    assert variable.formula_class.dated_formulas_class.__len__() == 0
+
+
+# 371 - end, one function without date
+
+class dated_attribute__one_formula(Variable):
+    column = IntCol
+    entity = Individu
+    definition_period = MONTH
+    label = u"Variable with end attribute, one function without date."
+    # start_date = datetime.date(1980, 1, 1)
+    end = '1989-12-31'
+
+    def formula(self, individu, period, nb):
+        return vectorize(self, nb)
+
+
+tax_benefit_system.add_variable(dated_attribute__one_formula)
+
+
+def test_call__dated_attribute__one_formula():
+    month = '1979-12'
+    simulation = new_simulation(tax_benefit_system, month)
+    assert simulation.calculate('dated_attribute__one_formula', month, extra_params=[100]) == 100
+
+    month = '1989-12'
+    simulation = new_simulation(tax_benefit_system, month)
+    assert simulation.calculate('dated_attribute__one_formula', month, extra_params=[100]) == 100
+
+    month = '1990-01'
+    simulation = new_simulation(tax_benefit_system, month)
+    result = simulation.calculate('dated_attribute__one_formula', month, extra_params=[100])
+    assert result == IntCol.default, result
+
+
+def test_dates__dated_attribute__one_formula():
+    variable = tax_benefit_system.column_by_name['dated_attribute__one_formula']
+    assert variable is not None
+    assert variable.end == '1989-12-31'
+
+    assert variable.formula_class.dated_formulas_class.__len__() == 1
+    formula = variable.formula_class.dated_formulas_class[0]
+    assert formula is not None
+    assert formula['start_instant'].date == datetime.date(1, 1, 1)
```

Comment:
Ok for tests. And it will show us the egyptian era case when it comes as `datetime.date.min` is not applied in `formulas.py` (to be consistent with formula name specifications).